segmentation_utils.py

```diff
@@ -6,6 +6,23 @@ def nothing(x):
     """Callback function for trackbars (does nothing)."""
     pass
 
+def resize_with_aspect_ratio(image, width=None, height=None, inter=cv2.INTER_AREA):
+    """Resize image while maintaining aspect ratio."""
+    dim = None
+    (h, w) = image.shape[:2]
+
+    if width is None and height is None:
+        return image
+
+    if width is None:
+        r = height / float(h)
+        dim = (int(w * r), height)
+    else:
+        r = width / float(w)

```

Comment:
Hi @TheCoderAdi , if you are converting the height to int, then also perform the same on width as well. Else there would be a minute difference in the size. Please perform the same on the if block as well.

Reply:
I have added the code , can you please review.